addons/survey/controllers/main.py

```diff
@@ -533,7 +533,7 @@ def survey_submit(self, survey_token, answer_token, **post):
             answer, comment = self._extract_comment_from_answers(question, post.get(str(question.id)))
             errors.update(question.validate_question(answer, comment))
             if not errors.get(question.id):
-                answer_sudo._save_lines(question, answer, comment, overwrite_existing=survey_sudo.users_can_go_back)
+                answer_sudo._save_lines(question, answer, comment, overwrite_existing=survey_sudo.users_can_go_back or question.save_as_nickname or question.save_as_email)
```

Comment:
Hello @amdi-odoo 

Fix looks good, however, can I ask why this is for 17.0?
I think the "users can go back, "save as nickname" and "save as email" features have been there for a very long time.

If the issue was introduced later with a different commit, can you add that commit's reference in the commit message?

Thanks for your work & cheers!